xdsl/dialects/fsm.py

```diff
@@ -0,0 +1,377 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    ContainerOf,
+    DictionaryAttr,
+    FlatSymbolRefAttr,
+    FunctionType,
+    IndexType,
+    IntegerType,
+    StringAttr,
+)
+from xdsl.ir import (
+    Attribute,
+    Block,
+    Operation,
+    Region,
+    SSAValue,
+)
+from xdsl.ir.core import Dialect
+from xdsl.irdl import (
+    AnyAttr,
+    AnyOf,
+    IRDLOperation,
+    attr_def,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    opt_operand_def,
+    region_def,
+    var_operand_def,
+    var_result_def,
+)
+from xdsl.traits import (
+    HasParent,
+    IsTerminator,
+    NoTerminator,
+    SymbolOpInterface,
+    SymbolTable,
+)
+from xdsl.utils.exceptions import VerifyException
+
+signlessIntegerLike = ContainerOf(AnyOf([IntegerType, IndexType]))
```

Comment:
We already have this [here](https://github.com/xdslproject/xdsl/blob/main/xdsl/dialects/arith.py#L40). IMO, it makes more sense to import that than redefining it. What do others think? @math-fehr @superlopuh

Reply:
I am confused, why is this back? :thinking:

Reply:
Oooh sorry, I tried to make the two PRs more similar as I made some major modifications to more branches yesterday and forgot about some of them! fixing it rn